tools/nd_freeze

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/perl
+#
+# Install access to the NeuroDebian snapshot repository that allows access
+# to old packages based on dates and version numbers.
+#
+# Usage: nd_freeze [--keep-original-apt-sources] date
+#
+# The UTC date/time format sent to NeuroDebian to get the repo info is in the
+# format yyyymmddThhmmssz or yyymmdd. However, the script will handle and
+# reformat a number of date/time formats. Quotes are required if there is a
+# space between the date and time components:
+#
+#   yyyymmddThh:mm:ssZ
+#   yyyymmdd
+#   mm/dd/yyyy
+#   yyyy-mm-dd
+#   yyyymmddThh:mm
+#   "yyyy-mm-dd hh:mm:ss"
+#   "mm/dd/yy hh:mm:ss"
+#
+# Browse available snapshots here:
+# http://snapshot-neuro.debian.net:5002/archive/neurodebian/
+#
+# If there is no import at the exact time you specified you will get both the
+# previous and next available timestamped repos of the one you specified to make
+# available versions that may have been placed in the repo before your
+# selected time but after the previous snapshot.
+#
+# Setting the command line switch, --keep-original-apt-sources will leave
+# the original, non-snapshot source active so that up-to-date packages can
+# still be installed.
+
+require 5.002;
+use Socket;
+
+
+# Function prints information to the screen.
+# 
+# Parameters
+# ----------
+# $output
+#     string : Text to print to the screen.
+# 
+sub info {
+    my ($output) = @_;
+    print "INFO: ${output}\n";
+}
+
+# Function gets the page contents of the provided URL.
+#
+# Parameters
+# ----------
+# $url
+#     string : URL of page to retrieve
+#
+# Returns
+# -------
+# string : Contents of URL page 
+sub get_www_content {
+    my ($url) = @_;
+    $url =~ /http:\/\/([^\/:]+):?(\d+)*(.*)/;
+    my $dest = $1;
+    my $port = $2;
+    my $file = $3;
+    $port = 80 if (!$port);
+    my $proto = getprotobyname('tcp');
+    socket(F, PF_INET, SOCK_STREAM, $proto);
+    my $sin = sockaddr_in($port,inet_aton($dest));
+    connect(F, $sin) || return undef;
+    my $old_fh = select(F);
+    $| = 1;
+    select($old_fh);
+    print F "GET $file HTTP/1.1\nHost: ${dest}\n\n";
+    $/ = undef;
+    $contents = <F>;
+    close(F);
+    return $contents;
+}
+
+
+# Function to run apt-get update
+sub run_apt_get_update {
+    info "Refreshing apt cache";
+    if (qx!grep ubuntu /etc/os-release!) {
+        print qx/apt-get update --no-allow-insecure-repositories/;
```

Comment:
Could those print become a debug function similar to info? So later on we could add option to enable or disable them